ports.go

```diff
@@ -26,140 +28,105 @@ type DevicePortServiceOp struct {
 
-func (i *DevicePortServiceOp) Bond(portID string, bulkEnable bool) (*Port, *Response, error) {
-	path := fmt.Sprintf("%s/%s/bond", portBasePath, portID)
-	if bulkEnable {
-		path += "?bulk_enable=true"
-	}
-	output := new(Port)
-
-	resp, err := i.client.DoRequest("POST", path, nil, output)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return output, resp, nil
+func (i *DevicePortServiceOp) Unassign(portID, vlanID string) (*Port, *Response, error) {
+	path := fmt.Sprintf("%s/%s/unassign", portBasePath, portID)
+	return i.assignmentAction(portID, vlanID, path)
 }
 
-type PortDisbondRequest struct {
-	PortID      string
-	BulkDisable bool
+func (i *DevicePortServiceOp) assignmentAction(portID, vlanID, path string) (*Port, *Response, error) {
+	req := PortRequest{
+		PortID:           portID,
+		VirtualNetworkID: vlanID,
+	}
+	return i.portAction(path, &req)
 }
 
-func (i *DevicePortServiceOp) Disbond(portID string, bulkDisable bool) (*Port, *Response, error) {
+func (i *DevicePortServiceOp) Bond(portID string, bulkEnable bool) (*Port, *Response, error) {
 	path := fmt.Sprintf("%s/%s/bond", portBasePath, portID)
-	if bulkDisable {
-		path += "?bulk_disable=true"
-	}
-	output := new(Port)
-
-	resp, err := i.client.DoRequest("POST", path, nil, output)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return output, resp, nil
+	req := BondRequest{PortID: portID, BulkEnable: bulkEnable}
+	return i.portAction(path, &req)
 }
 
-func (i *DevicePortServiceOp) GetBondedPort(deviceID string) (*Port, bool, error) {
-	device, _, err := i.client.Devices.Get(deviceID)
-	for _, port := range device.NetworkPorts {
-		if port.Type == "NetworkBondPort" {
-			return &port, true, nil
-		}
-	}
-
-	return nil, false, err
+func (i *DevicePortServiceOp) Disbond(portID string, bulkDisable bool) (*Port, *Response, error) {
+	path := fmt.Sprintf("%s/%s/disbond", portBasePath, portID)
+	req := DisbondRequest{PortID: portID, BulkDisable: bulkDisable}
+	return i.portAction(path, &req)
 }
 
-// Private helper methods
-func (i *DevicePortServiceOp) assignVirtualNetwork(input *PortAssignRequest) (*PortAssignResponse, *Response, bool, error) {
-	path := fmt.Sprintf("%s/%s/assign", portBasePath, input.PortID)
-	assignResponse := new(PortAssignResponse)
+func (i *DevicePortServiceOp) portAction(path string, req interface{}) (*Port, *Response, error) {
+	port := new(Port)
 
-	resp, err := i.client.DoRequest("POST", path, input, assignResponse)
+	resp, err := i.client.DoRequest("POST", path, req, port)
 	if err != nil {
-		return nil, resp, false, err
+		return nil, resp, err
 	}
 
-	return assignResponse, resp, true, err
+	return port, resp, err
 }
 
-func (i *DevicePortServiceOp) convertToLayerTwo(input *PortAssignRequest) (*PortAssignResponse, *Response, bool, error) {
-	path := fmt.Sprintf("%s/%s/convert/layer-2", portBasePath, input.PortID)
-	assignResponse := new(PortAssignResponse)
+func (i *DevicePortServiceOp) ConvertToLayerTwo(portID string) (*Port, *Response, error) {
```

Comment:
Breaking this out into another distinct function makes sense, but it then also means every application has to implement their own logic for deciding when a conversion to L2 is necessary before calling `Assign`. If this is going to stay a distinct function call, then there should be another function that is called `AssignAndConvert`

Reply:
The type 1E has L2 support on "eth1" interface, but doesn't need to be converted to Layer 2, I split it because of this.